### PR TITLE
Switch to `winpthreads-devel`

### PR DESCRIPTION
**Pull request by h-vetinari**
Now that there's a new [`libwinpthread`](https://github.com/conda-forge/m2w64-sysroot-feedstock) that has a `pthreads-win32 <0.0a0` constraint, there are conflicts because libhwloc is still built for the "old world", e.g.
```
Could not solve for environment specs
The following packages are incompatible
├─ libblas 3.9.0.* 26_win64_mkl is installable and it requires
│  └─ mkl 2025.0.0 hb752886_928, which requires
│     └─ tbb >=2022.0.0 , which requires
│        └─ libhwloc >=2.11.2,<2.11.3.0a0 , which requires
│           └─ pthreads-win32 with the potential options
│              ├─ pthreads-win32 2.9.1 would require
│              │  └─ vc 9.* , which does not exist (perhaps a missing channel);
│              ├─ pthreads-win32 2.9.1, which can be installed;
│              └─ pthreads-win32 2.9.1 would require
│                 └─ vc 10.* , which does not exist (perhaps a missing channel);
├─ libgcc 14.2.0.* h1383e82_1 is not installable because it requires
│  └─ libwinpthread >=12.0.0.r4.gg4f2fc60ca , which requires
│     └─ pthreads-win32 <0.0a0 , which conflicts with any installable versions previously reported;
├─ libgfortran5 14.2.0.* hf020157_1 is not installable because it requires
│  └─ libwinpthread >=12.0.0.r4.gg4f2fc60ca , which cannot be installed (as previously explained);
├─ libgomp 14.2.0.* h1383e82_1 is not installable because it requires
│  └─ libwinpthread >=12.0.0.r4.gg4f2fc60ca , which cannot be installed (as previously explained);
└─ libwinpthread 12.0.0.r4.gg4f2fc60ca h57928b3_8, which cannot be installed (as previously explained).
```

Switch to the newer implementation

### CODE REVIEW

**Comment by h-vetinari** [comment]
I modified the path according to the new [location](https://conda-metadata-app.streamlit.app/?q=conda-forge%2Fnoarch%2Fmingw-w64-ucrt-x86_64-winpthreads-git-12.0.0.r4.gg4f2fc60ca-h7428d3b_8.conda) in the sysroot, but it still fails:
```
clang: error: no such file or directory: 'D:/bld/libhwloc_1731117472764/_build_env/Library/x86_64-w64-mingw32/sysroot/usr/lib/libpthread.a'
```

**Comment by h-vetinari** [comment]
@conda-forge/libhwloc, this is ready (and necessary to unblock MKL in our blas metapackage), PTAL! :)

**Review by jan-janssen — APPROVED** [review]
Looks good to me

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,24 +36,21 @@ build:
...
     - autotools_clang_conda     # [win]
-    # Workaround for https://github.com/conda-forge/libhwloc-feedstock/pull/69#issuecomment-1739406306
-    - clang 16.*                # [win]
     - make                      # [unix]
     - automake                  # [unix]
     - libtool                   # [unix]
     - autoconf                  # [unix]
   host:
     - libxml2
-    - pthreads-win32            # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:50]: `- winpthreads-devel`

Comment (h-vetinari) [reply recipe/meta.yaml:50]: Both in host & in run, or just on host?

**Comment by isuruf** [comment]
@conda-forge-admin rerender

**Comment by h-vetinari** [comment]
This should be ready now.

**Comment by isuruf** [comment]
Thanks

**Pull request merged**